packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx

```diff
@@ -514,123 +534,131 @@ export const useTerraform = ({
     }
   };
 
-  const synth = () => {
-    React.useEffect(() => {
-      const invoke = async () => {
-        await execTerraformSynth(false);
-      };
-
-      invoke();
-    }, []);
-
-    return state;
+  return {
+    synth: execTerraformSynth,
+    deploy: execTerraformApply,
+    diff: execTerraformPlan,
+    output: execTerraformOutput,
+    destroy: execTerraformDestroy,
+    init: execTerraformInit,
   };
+};
 
-  const init = () => {
-    React.useEffect(() => {
-      const invoke = async () => {
-        await execTerraformSynth();
-      };
-      invoke();
-    }, []);
+const useRunOnce = <Fn extends (...args: any[]) => any>(
```

Comment:
Nitpick: We don't necessarily need to use any here :)

```suggestion
const useRunOnce = <T, TR, Fn extends (...args: T[]) => TR>(
```

Reply:
Ah, nice!

Reply:
Hm, that seems to be incomplete or am I missing something?

<img width="1122" alt="image" src="https://user-images.githubusercontent.com/1112056/124737486-48e35800-df18-11eb-80da-7ae04890960f.png">

Reply:
Hmm, seems like the type narrowing does not work as I thought it would, let's keep the anys for now